src/core/reaction_ensemble.hpp

```diff
@@ -109,7 +110,17 @@ struct DegreeOfAssociationCollectiveVariable : public CollectiveVariable {
 class ReactionAlgorithm {
 
 public:
-  ReactionAlgorithm() = default;
+  ReactionAlgorithm() = delete;
+  ReactionAlgorithm(int seed) {
```

Comment:
I think you should go for the delegation form here:

```c++
ReactionAlgorithm(int seed) : m_generator(std::seed_seq{seed}), 
                                               m_normal_distribution(0, 1), 
                                               m_uniform_real_distribution(0, 1) {
  m_generator.discard(1e6);
}
```

Also I think 0 and 1 are the default parameters for the distributions and are not needed.